src/node.js

```diff
@@ -328,22 +328,29 @@
     // Run callbacks that have no domain.
     // Using domains will cause this to be overridden.
     function _tickCallback() {
-      var callback, threw, tock;
+      var callback, args, tock;
 
       do {
         while (tickInfo[kIndex] < tickInfo[kLength]) {
           tock = nextTickQueue[tickInfo[kIndex]++];
           callback = tock.callback;
-          threw = true;
-          try {
-            if (tock.args === undefined)
-              callback();
-            else
-              callback.apply(null, tock.args);
-            threw = false;
-          } finally {
-            if (threw)
-              tickDone();
+          args = tock.args;
```

Comment:
a nit: would you mind adding a comment briefly explaining why this call is separated into a switch statement, for future readers?